.eslintrc.js

```diff
@@ -1,44 +1,67 @@
+/* global module */
 module.exports = {
-  root: true,
-  parserOptions: {
-    sourceType: 'module',
+  ignorePatterns: [
+    "!.prettierrc.js",
+    "**/!.eslintrc.js",
+    "**/dist*/",
+    "**/*__GENERATED__*",
+    "**/build",
+    "**/public",
+    "**/.cache",
+    "**/styles",
+  ],
+  parser: "@typescript-eslint/parser",
+  plugins: [
+    "@typescript-eslint",
+    "prettier",
+    // We enforce certain rules on how imports are handled
+    "import",
+  ],
+  extends: [
+    "eslint:recommended",
+    "plugin:prettier/recommended",
+    "plugin:import/errors",
+    "plugin:import/warnings",
+    "plugin:import/typescript",
+  ],
+  rules: {
+    "no-self-compare": "error",
+    "import/order": [
+      1,
+      {
+        groups: [
+          "builtin",
+          "external",
+          "internal",
+          "parent",
+          "sibling",
+          "index",
+        ],
+        "newlines-between": "always",
+      },
+    ],
+    /**
+     * The default setting for Prettier is 'warn' because then it shows as yellow squiggly lines
+     * in the VS Code IDE. However, it means `eslint` will not have an error code if there is warning
+     * due to prettier unles you also add the `--max-warnings=0` flag in front of it. So, in the `lint-staged`
+     * scripts in the packages within this monorepo, we add that flag so that the precommit hooks
```

Comment:
Do we understand this comment? Does this comment hold true for the `nitro-gui` repo?

Reply:
Removed in https://github.com/statechannels/nitro-gui/pull/19/commits/db5898edf20411442fd3c6d65cd6fd580d346218